src/main/java/ui/MenuControl.java

```diff
@@ -87,7 +87,35 @@ private Menu createFileMenu() {
         return file;
 
     }
-    
+
+    private void onBoardNew() {
+        logger.info("Menu: Boards > New");
+
+        if (confirmNewBoardCreation()) {
+            panels.closeAllPanels();
+            onBoardSaveAs();
+        } else {
+            logger.info("New board creation cancelled");
+        }
```

Comment:
Use guard conditions to deal with the low hanging fruit first. It reduces the complexity of the execution paths.
```java
if (!confirmNewBoardCreation()) {
    logger.info("New board creation cancelled");
    return;
}

panels.closeAllPanels();
onBoardSaveAs();
```